lib/polyfill/media_capabilities.js

```diff
@@ -217,6 +235,60 @@ shaka.polyfill.MediaCapabilities = class {
     return res;
   }
 
+  /**
+   * Checks if the given media parameters of the video or audio streams are
+   * supported by the Cast platform.
+   * @param {{
+   *   contentType: string,
+   *   width: (number|undefined),
+   *   height: (number|undefined),
+   *   frameRate: (number|undefined),
+   *   transferFunction: (string|undefined)
+   * }} options canCastDisplayType() options.
+   *     contentType: A valid MIME type and (optionally) a codecs parameter.
+   *     width: Describes the stream horizontal resolution in pixels.
+   *     height: Describes the stream vertical resolution in pixels.
+   *     frameRate: Describes the frame rate of the stream.
+   *     transferFunction: Describes the video transfer function supported by
+   *         the rendering capabilities of the user agent.
+   * @return {boolean} `true` when the stream can be displayed on a Cast device.
+   * @private
+   */
+  static canCastDisplayType_({
+    contentType,
+    width = undefined,
+    height = undefined,
+    frameRate = undefined,
+    transferFunction = undefined,
+  }) {
+    if (!(window.cast && cast)) {
+      shaka.log.error('Expected cast namespace to be available!');
+      throw new shaka.util.Error(
+          shaka.util.Error.Severity.CRITICAL,
+          shaka.util.Error.Category.CAST,
+          shaka.util.Error.Code.CAST_API_UNAVAILABLE);
+    } else if (!(cast.__platform__ && cast.__platform__.canDisplayType)) {
+      shaka.log.warning('Expected cast APIs to be available! Falling back to ' +
```

Comment:
This is very strange.  I'd like to help you look into that, but I don't want to hold up the PR for it.

Reply:
Yeah, was held up on this for a while - the lab tests were failing without this check.

Reply:
I spent a solid day on this, and it's time for me to give up for now.  I managed to forward the Cast platform APIs through an iframe boundary, but not in a way that is compatible with Karma's websocket usage.  :-(